lib/concur/thread_pool.fz

```diff
@@ -118,6 +118,46 @@ is
     bq.enqueue task
 
 
+  public submit(T type, task ()->T) option (concur.Future T) =>
```

Comment:
I agree with Simon. One way to achieve this would be to add a `pre !bq.is_closed` and, since `shutdown` may be done concurrently, a `panic` in case `bq.enqueue` returns `false`.

Also, please add a comment explaining the behavior in  these corner cases.

Reply:
If I add `pre !bq.is_closed` I have to change visibility of bq. Another option might be to introduce an `is_shutdown` feature in `thread_pool` but that is for another PR.